pkg/etcd/client.go

```diff
@@ -149,8 +158,10 @@ func (c *Client) Txn(ctx context.Context, cmps []clientv3.Cmp, opsThen, opsElse
 // Grant delegates request to clientv3.Lease.Grant
 func (c *Client) Grant(ctx context.Context, ttl int64) (resp *clientv3.LeaseGrantResponse, err error) {
 	err = retryRPC(EtcdGrant, c.metrics[EtcdGrant], func() error {
+		grantCtx, cancel := context.WithTimeout(ctx, etcdClientTimeoutWithoutRetry)
```

Comment:
Should we use `etcdClientTimeoutWithRetry` here?

Reply:
We need to ensure that it returns in a reasonable amount of time, otherwise the operation may never retry due to [blocking](https://github.com/pingcap/tiflow/blob/ce83af5ad9c2ad148aab73ca620367d9cfc6fca0/pkg/etcd/client.go#L96).